In [12]:
import pandas as pd
import numpy as np
import statistics

In [30]:
df = pd.DataFrame()
df['name'] = ['Greg','Marcia','Peter','Jan','Bobby','Cindy','Oliver']
df['age'] = [14,12,11,10,8,6,8]
df

name  age
0    Greg   14
1  Marcia   12
2   Peter   11
3     Jan   10
4   Bobby    8
5   Cindy    6
6  Oliver    8

1. Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?

In [31]:
print('Mean: ', np.mean(df['age']))
print('Median: ',np.median(df['age']))
print('Mode: ', statistics.mode(df['age']))

Mean:  9.857142857142858
Median:  10.0
Mode:  8


In [32]:
df.describe()

age
count   7.000000
mean    9.857143
std     2.734262
min     6.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

In [37]:
print('Variance: ', np.var(df.age))
print('Standard Deviation: ', np.std(df.age))
print('Standard Error: ',np.std(df.age)/np.sqrt(len(df['age']-1)))

Variance:  6.408163265306122
Standard Deviation:  2.531435020952764
Standard Error:  0.9567925036515135


2. Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

As mean and median are close to each other, means there is not outlier in this dataset, mean and standard deviation are good to make the estimation. 

3. Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

In [39]:
df['age'] = [14,12,11,10,8,7,8]

In [40]:
print('Mean: ', np.mean(df['age']))
print('Median: ',np.median(df['age']))
print('Mode: ', statistics.mode(df['age']))
print('Variance: ', np.var(df.age))
print('Standard Deviation: ', np.std(df.age))
print('Standard Error: ',np.std(df.age)/np.sqrt(len(df['age']-1)))

Mean:  10.0
Median:  10.0
Mode:  8
Variance:  5.428571428571429
Standard Deviation:  2.32992949004287
Standard Error:  0.8806305718527109


Only median and mode didn't change, others all changed.

4. Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?


In [42]:
df1 = [14,12,11,10,8,7,1]

In [45]:
print('Mean: ', np.mean(df1))
print('Median: ',np.median(df1))
print('There is no mode in this data.')
print('Variance: ', np.var(df1))
print('Standard Deviation: ', np.std(df1))
print('Standard Error: ',np.std(df1)/np.sqrt(len(df1)-1))

Mean:  9.0
Median:  10.0
There is no mode in this data.
Variance:  15.428571428571429
Standard Deviation:  3.927922024247863
Standard Error:  1.6035674514745466


As change the last number from 8 to 1 makes the mean decreased and also 1 is unreasonable in this case, median is a better choice to descrive the central tendency.  

5. On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

In [46]:
list = [20,23,17]
np.mean(list)

20.0

As SciPhi Phanatic only shows 5% fans, which is an outlier in this dataset, also the topic of this magazine is not quite related to our subject, it tends to bring bias into the data. If only consider the other three samples, the estimation would be 20%.

In [52]:
list1 = [20,23,17,5]
print('The range of average number is between ',np.mean(list1)-np.std(list1)/np.sqrt(3),' and ',np.mean(list1)+np.std(list1)/np.sqrt(3))

The range of average number is between  12.305066540485125  and  20.194933459514875


If using standard error with average value of the data, we can also say the range of estimation is between  12.305066540485125  and  20.194933459514875.